gems/aws-sdk-core/lib/aws-sdk-core/plugins/regional_endpoint.rb

```diff
@@ -150,7 +150,18 @@ def resolve_endpoint(cfg)
           # that a custom endpoint has NOT been configured by the user
           cfg.override_config(:regional_endpoint, true)
 
-          resolve_legacy_endpoint(cfg)
+          # preserve legacy (pre EP2) client.config.endpoint still resolves a value
+          # when needed.
+          struct = cfg.instance_variable_get(:@struct)
```

Comment:
I don't understand this - and it's very implementation specific. When would this thing not exist?

Reply:
Was being defensive here.  If I add an accessor for struct to ConfigResolver, then I can skip checking probably